library/constant_time.c

```diff
@@ -514,6 +514,11 @@ int mbedtls_ct_hmac( mbedtls_svc_key_id_t key,
     PSA_CHK( psa_hash_update( &operation, add_data, add_data_len ) );
     PSA_CHK( psa_hash_update( &operation, data, min_data_len ) );
 
+    /* Clear the hash buffer in advance, thus preventing leaving
+     * unintialised data in there which could be interpreted
+     * as a valid hash in case of error. */
```

Comment:
(side note) all zeros is a perfectly valid hash result, just very unlikely. the difference is that random uninitialised garbage might look like a reasonable hash result to a human debugging things that aren't working, whereas all 0s will suggest to the human that the hash output wasn't written.

not sure how best to capture that in a comment, though!

Reply:
Here's how it's worded in `mbedtls_psa_hash_finish`. PSA code uses a memset of `'!'` rather than 0 to specifically indicate “if you're seeing this in output, you've done something wrong”.

    /* Fill the output buffer with something that isn't a valid hash
     * (barring an attack on the hash and deliberately-crafted input),
     * in case the caller doesn't check the return status properly. */
        memset( hash, '!', hash_size );